script.js

```diff
@@ -0,0 +1,908 @@
+      return isAnyKindOrPair;
+    }
+  }
+};
+
+// Function that checks for a flush recursively,
+// taking in playerHand.length-1 initially
+const checkForFlush = (numCardsToCheck) => {
+  let i = numCardsToCheck;
+  if (i > 1) {
+    if (playerHand[i].suit === playerHand[i - 1].suit) {
+      i -= 1;
+      return checkForFlush(i);
+    }
+    // If any 2 consecutive cards'suits don't match then immediately break
+    // and declare that there is no flush;
+    isFlush = false;
+    return isFlush;
+  }
+  // Once we reached the base case (first 2 cards in playerHand array),
+  // check if the 2 cards are the same
+  if (playerHand[i].suit === playerHand[i - 1].suit) {
+    nameOfWinCombi = 'Flush!';
+    rankOfHand = 4;
+    isFlush = true;
+    return isFlush;
+  }
+};
+
+// Function that checks for straight-combi
+const checkForStraight = () => {
+  // Sort hand by descending order by rank (use spread operator to prevent original hand array
+  //  from changing)
+  const sortedHand = [...playerHand].sort((a, b) => b.rank - a.rank);
+  console.log(playerHand, 'playerHand');
+
+  // loop through each element and check if difference between each is
+  // exactly one...
+  let countOfConsecutiveRankCards = 1;
+  for (let i = 0; i < playerHand.length - 1; i += 1) {
+    if (sortedHand[i].rank - sortedHand[i + 1].rank === 1) {
+      countOfConsecutiveRankCards += 1;
+    }
+  }
+  // For checking if 1st 2 card is K and Q first
+  const highestCardName = sortedHand[0].name;
+  const secondHighestCardName = sortedHand[1].name;
+
+  if (highestCardName === 'King' && secondHighestCardName === 'Queen' && countOfConsecutiveRankCards >= 4) {
+    // Check if last card in the sorted hand is an ace
+    // because its rank is 0, it will not show up as a consecutive card
+    if (sortedHand[sortedHand.length - 1].name === 'Ace') {
+      nameOfWinCombi = 'Ace-high Straight';
+      rankOfHand = 5;
+      isStraight = true;
+    }
+  } else if (countOfConsecutiveRankCards === sortedHand.length) {
+    nameOfWincombi = `${highestCardName}-high Straight`;
+    rankOfHand = 5;
+    isStraight = true;
+  }
+};
+
+// Function that checks for a straight flush
+const checkForStraightFlush = () => {
+  if (isFlush === true && isStraight === true) {
+    isStraightFlush = true;
+    nameOfWinCombi = 'Straight Flush!';
+    rankOfHand = 1;
+  }
+};
+
+// Function that resets all global trackers /(variables) of any winning combi
+const resetAllWinningCombiStatus = () => {
+  isAnyKindOrPair = false;
+  isFullHouse = false;
+  isFlush = false;
+  isStraight = false;
+  isStraightFlush = false;
+  // reset the array that tracks number of common cards
+  // per iteration of checks
+  numCommonCardsArray.length = 0;
+  console.log('status is reset');
+};
+
+// Function that checks hand for any winning combination
+const checkForWinCombi = () => {
+  compareCardRank();
+  checkForKindsnPairs();
+  checkForStraight();
+  checkForFlush(playerHand.length - 1);
```

Comment:
Strange that only this function requires an input. Since `playerHand` is a global, why is it necessary to pass `playerHand`'s attributes as a parameter?

Reply:
I can't remember the exact reason why, but I might have put it in as a reminder of what arguments checkForFlush takes in